packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/mappers.ts

```diff
@@ -46,16 +53,64 @@ export const toNetworkInfo = ({
   }
 });
 
+export const toLedgerTip = ({ block_no, slot_no, hash }: LedgerTipModel): Cardano.Tip => ({
+  blockNo: block_no,
+  hash: Cardano.BlockId(hash.toString('hex')),
+  slot: slot_no
+});
+
+export const toWalletProtocolParams = ({
+  coins_per_utxo_word,
+  max_tx_size,
+  max_val_size,
+  max_collateral_inputs,
+  min_pool_cost,
+  pool_deposit,
+  key_deposit,
+  protocol_major,
+  protocol_minor,
+  min_fee_a,
+  min_fee_b
+}: WalletProtocolParamsModel): ProtocolParametersRequiredByWallet => ({
+  coinsPerUtxoWord: Number(coins_per_utxo_word),
+  maxCollateralInputs: max_collateral_inputs,
+  maxTxSize: max_tx_size,
+  maxValueSize: Number(max_val_size),
+  minFeeCoefficient: min_fee_a,
+  minFeeConstant: min_fee_b,
+  minPoolCost: Number(min_pool_cost),
+  poolDeposit: Number(pool_deposit),
+  protocolVersion: {
+    major: protocol_major,
+    minor: protocol_minor
+  },
+  stakeKeyDeposit: Number(key_deposit)
+});
+
+export const toGenesisParams = (genesis: GenesisData): Cardano.CompactGenesis => ({
+  ...genesis,
+  systemStart: new Date(genesis.systemStart)
+});
+
 export const loadGenesisData = async (cardanoNodeConfigPath: string): Promise<GenesisData> => {
   try {
     const genesisFilePath = require(path.resolve(cardanoNodeConfigPath)).ShelleyGenesisFile;
     const genesis = JSONbig({ useNativeBigInt: true }).parse(
       fs.readFileSync(path.resolve(path.dirname(cardanoNodeConfigPath), genesisFilePath), 'utf-8')
     );
+
     return {
+      activeSlotsCoefficient: genesis.activeSlotsCoeff,
+      epochLength: genesis.epochLength,
+      maxKesEvolutions: genesis.maxKESEvolutions,
       maxLovelaceSupply: genesis.maxLovelaceSupply,
```

Comment:
This actually highlights an opportunity to improve our `NetworkInfoProvider` interface, by exposing `maxLovelaceSupply` and `networkId` under `genesisParameters`, which is already part of the `CompactGenesis` type. That would leave `NetworkInfo.network.timeSettings`, which we can consider lifting as a new top-level provider method, and then remove the `NetworkInfo` type by hoisting the remaining properties too i.e.

**Before**
``` ts
export type NetworkInfo = {
  network: {
    id: Cardano.NetworkId;
    magic: Cardano.NetworkMagic;
    timeSettings: TimeSettings[];
  };
  lovelaceSupply: AssetSupply;
  stake: StakeSummary;
};

export interface NetworkInfoProvider {
  currentWalletProtocolParameters(): Promise<ProtocolParametersRequiredByWallet>;
  genesisParameters(): Promise<Cardano.CompactGenesis>;
  ledgerTip(): Promise<Cardano.Tip>;
  networkInfo(): Promise<NetworkInfo>;
}
```

**After**
``` ts
export interface NetworkInfoProvider {
  currentWalletProtocolParameters(): Promise<ProtocolParametersRequiredByWallet>;
  genesisParameters(): Promise<Cardano.CompactGenesis>;
  lovelaceSupply(): Promise<AssetSupply>;
  stake(): Promise<StakeSummary>;
  timeSettings(): Promise<TimeSettings[]>;
}
```

Reply:
Agree, should probably be out of scope of this PR though

Reply:
I think we should at least do the genesis consolidation here, as right now it's a bit confusing on this aspect.

Reply:
To be done in ADP-1911